samples: matter: Add support for migrating DAC to KMU fo NRF54L (#18104)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

  * The :ref:`ug_matter_device_memory_profiling` section to the :ref:`ug_matter_device_optimizing_memory` page.
    The section contains useful commands for measuring memory and troubleshooting tips.
  * The ZMS file subsystem to all devices that contain RRAM, such as the nRF54L Series devices.

>   * Migration of the Device Attestation Certificates private key to Key Management Unit (KMU) for nRF54L SoC series.

Reviewer: ```suggestion
  * Migration of the Device Attestation Certificates private key to Key Management Unit (KMU) for the nRF54L Series SoCs.
```

    The new Kconfig option now works for both NVS and ZMS file system backends.
  * The firmware version format used for informational purposes when using the :file:``VERSION`` file.
    The format now includes the optional ``EXTRAVERSION`` component.

>   * Enabled storing the Device Attestation Certificates private key in the Trusted Storage library for all platforms that support the PSA crypto API.

Reviewer: ```suggestion
  * Storing the Device Attestation Certificates private key in the Trusted Storage library to be enabled for all platforms that support the PSA crypto API.
```

>     See :ref:`matter_platforms_security_dac_priv_key_its`.

Reviewer: ```suggestion
    See :ref:`matter_platforms_security_dac_priv_key_its` for more information.
```

    * Added :ref:`Matter Lock schedule snippet <matter_lock_snippets>`, and updated the documentation to use the snippet.

* :ref:`matter_template_sample` sample:

>     * Set the DAC private key migration from factory data to KMU for the nRF54L SoC series by default.

Reviewer: ```suggestion
    * Updated the DAC private key migration from factory data to KMU to be enabled for the nRF54L Series SoCs by default.
```
